src/common/hooks/use-random-tx.tsx

```diff
@@ -1,19 +1,25 @@
 import { useCallback } from 'react';
 import { navgiateToRandomTx } from '@common/utils';
 import { useToast } from '@common/hooks/use-toast';
+import { useSelector } from 'react-redux';
+import { selectCurrentNetworkUrl } from '@store/ui/selectors';
+import { RootState } from '@store';
 
 export const useNavigateToRandomTx = () => {
+  const { apiServer } = useSelector((state: RootState) => ({
+    apiServer: selectCurrentNetworkUrl(state),
+  }));
   const { addCriticalToast } = useToast();
   const handleRandomTxClick = useCallback(async () => {
     try {
-      await navgiateToRandomTx();
+      await navgiateToRandomTx(apiServer as string)();
```

Comment:
Why is type casing needed here? Is it sometimes undefined? If possible, setting a default in store so it's always defined would be nicer than casting.